desktop.c

```diff
@@ -48,25 +48,43 @@ i18n_init(void)
 	}
 
 	/* ll_CC */
-	strncpy(llcc, p, sizeof(llcc));
+	int ret = snprintf(llcc, sizeof(llcc), "%s", p);
+	if (ret < 0) {
```

Comment:
@01micko Technically, you should also check for `ret >= sizeof(ret)` (string is truncated)

Reply:
Ok, but is `llcc[sizeof(llc) - 1] = '\0';` dissimilar to my original `strncpy(llcc, p, sizeof(llcc) - 1);` https://github.com/labwc/labwc-menu-generator/issues/19#issue-2354390332

Although I think I'd prefer 
```c
if ((ret < 0) || (ret >= sizeof(ret))) {
	return;
}

```

Reply:
The compiler warngs because

```
If there is no null byte among the first n bytes of src, the string placed in dest will not be null-terminated. 
```

... so the warning should go away if you ensure the string is terminated even if the buffer isn't big enough.

Reply:
I've marked this as a draft so I'll leave the choice up to @johanmalm

Reply:
Okay - I shouldn't have suggested `snprintf()`. I'd be cleaner + less controversial + will attract less compiler warnings if we just heap-allocate. Suggest we do some `xasprintf()` variant. Could use `g_strdup_printf()` with an `x` wrapper to make it die on bad allocation.

I'm running out of time tonight, but can do a patch if you prefer.

Reply:
Probably best if I leave it to you.